src/sentry/api/endpoints/release_thresholds/health_checks/is_new_issues_count_healthy.py

```diff
@@ -0,0 +1,47 @@
+from __future__ import annotations
+
+from logging import getLogger
+
+from sentry.api.endpoints.release_thresholds.types import EnrichedThreshold
+from sentry.models.group import Group
+from sentry.models.groupenvironment import GroupEnvironment
+from sentry.models.release_threshold.constants import TriggerType
+
+logger = getLogger("sentry.releases.servicer")
+
+
+def is_new_issues_count_healthy(release_threshold: EnrichedThreshold) -> bool:
+    release_id = release_threshold["release_id"]
+    project_id = release_threshold["project_id"]
+    time_range = (release_threshold["start"], release_threshold["end"])
+    try:
+        if release_threshold["environment"]:
+            new_issues = GroupEnvironment.objects.filter(
+                first_release__id=release_id,
```

Comment:
actually - since we have the reference to the release here, we can make this `first_release__version=release.version`

Reply:
Is that really safe though? Can we guarantee `release.version` is unique across all releases? You could have 2 different releases with the same "version" value in the column, I don't see a unique constraint https://github.com/getsentry/sentry/blob/dcfcfe63af2281f9936b6edf4cb754a9ee3862f2/src/sentry/models/release.py#L480

Reply:
Fair question

I'm not entirely sure - but i know our customers won't have access to their release id's and they'll be querying our APIs with their release versions instead.

So no matter what, this will be problematic if we're unable to identify a particular release with just the version

Reply:
Sure, but why are we trying to compromise on the query here? If we have access to the ids, let's use the ids. If/when the contract can finally start using ids, that won't be an issue, and this code won't have to change.

Reply:
🤔 

hmmm IMO we aren't compromising on the query at all.

our api's will never take ids because users should never want nor need access to our internal ids.
Say for example - you're releasing your apps and you're tagging your versions/deploys with semver versions. You go through your ci with `frontend@1.2.3` or something. Your ci/cd flow takes that tag ships it to jenkins/sentry/github/etc.

Now they want to check the health of their deploy. No user will ever be keeping reference to what versions relate to our sentry ids

So while we may not have a unique restriction on our models (not sure why we don't), if a user has non unique releases, then there's probably going to be more issues no matter what 🤷